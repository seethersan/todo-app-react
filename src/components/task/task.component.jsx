import React, { useContext } from "react";
import { List, Checkbox } from "antd";
import TasksDispatch from "../../hooks/tasks/task.provider";
import { completeTask, remove } from "../../hooks/tasks/task.action";

export default function TaskItem(props) {
    const dispatch = useContext(TasksDispatch);

    const {
        task: { id, title, isDone }
    } = props;

    const handleStatus = () => {
        dispatch(completeTask(id));
    };

    const handleRemove = () => {
        dispatch(remove(id));
    };

    return (
        <List.Item
          data-item={id}
          className={isDone ? "done" : "todo"}
          actions={[
            <a href="#" onClick={handleRemove}>
              Remove
            </a>
          ]}
        >
          <List.Item.Meta
            title={title}
            avatar={<Checkbox onChange={handleStatus} />}
          />
        </List.Item>
      );
}