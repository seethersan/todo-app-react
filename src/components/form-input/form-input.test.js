import * as react from 'react'
import { shallow, mount, render } from 'enzyme';
import { add } from "../../hooks/tasks/task.action";
import TaskForm from './form-input.component';

window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    };
};

it('expect to render TaskForm component', () => {
    expect(shallow(<TaskForm />)).toMatchSnapshot();
});

it('correctly add task', () => {
    const mockDispatch = jest.fn();

    jest.spyOn(react, 'useContext').mockImplementation(() => mockDispatch);

    const newTask = 'new task 1';
    
    const wrapper = mount(<TaskForm />);

    wrapper.find('input.form-input').simulate("change", { target: { value: newTask }});
    wrapper.find('button.form-button').simulate('click');

    expect(mockDispatch).toHaveBeenCalledWith(add(newTask))
});