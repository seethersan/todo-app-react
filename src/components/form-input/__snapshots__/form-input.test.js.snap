// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render TaskForm component 1`] = `ShallowWrapper {}`;
